src/mdformat/_cli.py

```diff
@@ -303,6 +303,32 @@ def make_arg_parser(
             plugin.add_cli_argument_group(group)
             for action in group._group_actions:
                 action.dest = f"plugin.{plugin_id}.{action.dest}"
+                if type(action) in {
+                    argparse._StoreTrueAction,
+                    argparse._StoreFalseAction,
+                }:
+                    import warnings
+
+                    is_store_true = type(action) is argparse._StoreTrueAction
+                    plugin_name = (
+                        action.container.title
+                        if hasattr(action, "container")
+                        else str(plugin)
+                    )
+                    text = (
+                        f"For {action.option_strings} from {plugin_name},"
+                        " the default will always override a value configured"
+                        " in TOML. To resolve, replace `.add_argument(...,"
+                        f' action="store_{str(is_store_true).lower()}")` with'
+                        f' `(..., action="store_const", const={is_store_true})`'
+                    )
```

Comment:
I think just recommending to add `default=None` or `default=argparse.SUPPRESS` would work regardless of the action used. Should also work with `action=store_true/false`, so I don't think there's even a need to mention them by name (I know I said otherwise in an issue comment, sorry :smile: )